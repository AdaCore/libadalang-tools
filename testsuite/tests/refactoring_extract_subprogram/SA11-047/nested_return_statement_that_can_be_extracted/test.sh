extract_subprogram -P default.gpr -S my_package.adb -SL 6 -SC 8 -EL 6 -EC 9 -N Foo
