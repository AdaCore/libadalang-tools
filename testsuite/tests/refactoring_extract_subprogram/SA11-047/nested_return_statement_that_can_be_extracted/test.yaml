title: nested_return_statement_that_can_be_extracted

description: extract_subprogram tool test

driver: shell_script
