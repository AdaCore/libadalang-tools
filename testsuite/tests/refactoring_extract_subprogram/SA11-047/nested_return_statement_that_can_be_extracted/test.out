Extracting as a ADA_SUBP_KIND_PROCEDURE:
my_package.adb
5:1-5:1 
      procedure Foo;

      procedure Foo is
      begin
         if I = 1 then
            declare
               function My_Nested_Function return Boolean is
               begin
                  return True;
               end My_Nested_Function;
            begin
               B := My_Nested_Function;
            end;
         end if;
      end Foo;


6:7-15:14 Foo;

